pkg/dockerregistry/server/app.go

```diff
@@ -55,16 +57,44 @@ type App struct {
 
 	// cache is a shared cache of digests and descriptors.
 	cache cache.DigestCache
+
+	// repositoryEnumerator allows to enumerate repository names known to registry
+	repositoryEnumerator RepositoryEnumerator
+}
+
+type registry struct {
+	registry   distribution.Namespace
+	enumerator RepositoryEnumerator
+}
+
+func (r *registry) Scope() distribution.Scope {
+	return r.registry.Scope()
+}
+
+func (r *registry) Repository(ctx context.Context, name reference.Named) (distribution.Repository, error) {
+	return r.registry.Repository(ctx, name)
+}
+
+func (r *registry) Repositories(ctx context.Context, repos []string, last string) (n int, err error) {
```

Comment:
Also a good candidate for metrics. Suitable as a follow-up after this and #79 are merged.